* Issue baking a new layer for a custom kernel #kernel
@ 2021-08-19 16:02 nagesh shamnur
  2021-09-01 17:29 ` [yocto] " Khem Raj
  0 siblings, 1 reply; 2+ messages in thread
From: nagesh shamnur @ 2021-08-19 16:02 UTC (permalink / raw)
  To: yocto

[-- Attachment #1: Type: text/plain, Size: 3857 bytes --]

Hi Group,
I am trying to add a new layer for a custom RTOS which supports RISCV32 architecture. When i run bitbake i bumped into following error:
"
ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: Variable RDEPENDS is set as not being package specific, please fix this. [pkgvarcheck]
ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: Variable RRECOMMENDS is set as not being package specific, please fix this. [pkgvarcheck]
ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: Fatal QA errors found, failing task.
ERROR: Failed to parse recipe: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb
WARNING: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-yappi_1.0.bb: Cooker received SIGTERM, shutting down...
ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: Variable RDEPENDS is set as not being package specific, please fix this. [pkgvarcheck]
ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: Variable RRECOMMENDS is set as not being package specific, please fix this. [pkgvarcheck]
ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: Fatal QA errors found, failing task.
WARNING: Cooker received SIGTERM, shutting down...
WARNING: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zipp_0.6.0.bb: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...
WARNING: Cooker received SIGTERM, shutting down...

Summary: There were 12 WARNING messages shown.
Summary: There were 7 ERROR messages shown, returning a non-zero exit code.
"

Looking for the exact line which might have created a problem, found out that the issue is happening in the following line
"PACKAGES =. "${PN}-test " from the file ./meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb.

i have tried the following options, but none of them have helped in solving the issue.
1) I suspected that missing python3native might have resulted in the problem and included "inherit python3native" in my recipe file,
2) Also checked for presence of RDEPENDS in my recipe file as below: "LAYERDEPENDS_xxx ="core meta-python"

Please suggest a way out of this issue.

Thanks!

[-- Attachment #2: Type: text/html, Size: 4032 bytes --]

^ permalink raw reply	[flat|nested] 2+ messages in thread
* Re: [yocto] Issue baking a new layer for a custom kernel #kernel
  2021-08-19 16:02 Issue baking a new layer for a custom kernel #kernel nagesh shamnur
@ 2021-09-01 17:29 ` Khem Raj
  0 siblings, 0 replies; 2+ messages in thread
From: Khem Raj @ 2021-09-01 17:29 UTC (permalink / raw)
  To: nagesh shamnur; +Cc: Yocto-mailing-list

ensure that you are using same release branches for all layer repos
generally RDEPENDS and RRECOMMENDS should be replaced with
RDEPENDS_<pkgname> etc.

On Wed, Sep 1, 2021 at 7:22 AM nagesh shamnur <nagesh.shamnur@gmail.com> wrote:
>
> Hi Group,
>    I am trying to add a new layer for a custom RTOS which supports RISCV32 architecture. When i run bitbake i bumped into following error:
> "
> ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: Variable RDEPENDS is set as not being package specific, please fix this. [pkgvarcheck]
> ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: Variable RRECOMMENDS is set as not being package specific, please fix this. [pkgvarcheck]
> ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb: Fatal QA errors found, failing task.
> ERROR: Failed to parse recipe: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb
> WARNING: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-yappi_1.0.bb: Cooker received SIGTERM, shutting down...
> ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: Variable RDEPENDS is set as not being package specific, please fix this. [pkgvarcheck]
> ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: QA Issue: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: Variable RRECOMMENDS is set as not being package specific, please fix this. [pkgvarcheck]
> ERROR: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-werkzeug_1.0.0.bb: Fatal QA errors found, failing task.
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: /home/gitee-ohos/ohos/AllScenariosOS/sources/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-zipp_0.6.0.bb: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
> WARNING: Cooker received SIGTERM, shutting down...
>
> Summary: There were 12 WARNING messages shown.
> Summary: There were 7 ERROR messages shown, returning a non-zero exit code.
> "
>
> Looking for the exact line which might have created a problem, found out that the issue is happening in the following line
> "PACKAGES =. "${PN}-test " from the file ./meta-openembedded/meta-python/recipes-devtools/python/python3-zopeinterface_4.7.1.bb.
>
> i have tried the following options, but none of them have helped in solving the issue.
> 1) I suspected that missing python3native might have resulted in the problem and included "inherit python3native" in my recipe file,
> 2) Also checked for presence of RDEPENDS in my recipe file as below: "LAYERDEPENDS_xxx ="core meta-python"
>
> Please suggest a way out of this issue.
>
> Thanks!
>
>
> 
>

^ permalink raw reply	[flat|nested] 2+ messages in thread
end of thread, other threads:[~2021-09-01 17:29 UTC | newest]

Thread overview: 2+ messages (download: mbox.gz / follow: Atom feed)
-- links below jump to the message on this page --
2021-08-19 16:02 Issue baking a new layer for a custom kernel #kernel nagesh shamnur
2021-09-01 17:29 ` [yocto] " Khem Raj

